CMakeLists.txt

```diff
@@ -188,50 +166,33 @@ foreach (OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES})
 endforeach()
 
 enable_testing()
-add_subdirectory(utest)
-
-if (NOT MSVC)
-	#only build shared library for MSVC
 
-	add_library(${OpenBLAS_LIBNAME}_static STATIC ${LA_SOURCES} ${LAPACKE_SOURCES} ${TARGET_OBJS})
-	set_target_properties(${OpenBLAS_LIBNAME}_static PROPERTIES OUTPUT_NAME ${OpenBLAS_LIBNAME})
-	set_target_properties(${OpenBLAS_LIBNAME}_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
-
-	if(SMP)
-	target_link_libraries(${OpenBLAS_LIBNAME} pthread)
-	target_link_libraries(${OpenBLAS_LIBNAME}_static pthread)
+if (USE_THREAD)
+  # Add threading library to linker
+  find_package(Threads)
+  if (THREADS_HAVE_PTHREAD_ARG)
+    set_property(TARGET ${OpenBLAS_LIBNAME} PROPERTY COMPILE_OPTIONS "-pthread")
+    set_property(TARGET ${OpenBLAS_LIBNAME} PROPERTY INTERFACE_COMPILE_OPTIONS "-pthread")
+  endif()
+  target_link_libraries(${OpenBLAS_LIBNAME} ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
-#build test and ctest
-add_subdirectory(test)
-if(NOT NO_CBLAS)
-add_subdirectory(ctest)
-endif()
+if (NOT MSVC AND NOT NOFORTRAN)
+  add_subdirectory(utest)
+  # Build test and ctest
+  add_subdirectory(test)
+  if(NOT NO_CBLAS)
+    add_subdirectory(ctest)
+  endif()
+elseif(MSVC)
```

Comment:
@isuruf That's correct. utest was failing when fortran was disabled. This change allows build to succeed.